File: lib/clusterbuster/pod_files/uperf-server.pl
#!/usr/bin/perl

use POSIX;
use strict;
my ($dir) = $ENV{'BAK_CONFIGMAP'};
require "$dir/clientlib.pl";

$SIG{TERM} = sub { POSIX::_exit(0); };
my ($listen_port) = parse_command_line(@ARGV);

sub runit() {
    timestamp("Starting uperf server on port $listen_port");
    system("uperf", "-s", "-v", "-P", "$listen_port");
    timestamp("Done!");
}

run_workload(1, \&runit);

File: lib/clusterbuster/pod_files/uperf-server.py
#!/usr/bin/env python3

import subprocess
import time
from clusterbuster_pod_client import clusterbuster_pod_client


class uperf_server_client(clusterbuster_pod_client):
    """
    Server side of uperf test for clusterbuster
    """

    def __init__(self):
        try:
            super().__init__()
            self.listen_port = self._args[0]
        except Exception as err:
            self._abort(f"Init failed! {err} {' '.join(self._args)}")

    def runit(self, process: int):
        self._timestamp(f"Starting uperf server on port {self.listen_port}")
        while subprocess.run(['uperf', '-s', '-v', '-P', self.listen_port]).returncode != 0:
            self._timestamp("uperf server failed to start: retrying in 10 seconds")
            time.sleep(10)
        self._timestamp("Done!")


uperf_server_client().run_workload()
